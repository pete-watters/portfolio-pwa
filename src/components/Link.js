import React, { memo } from 'react';
import PropTypes from 'prop-types';

const Link = ({href, text}) =>
    <a href={href} target="_blank" rel="noopener noreferrer">{text}</a>;

Link.propTypes = {
  href: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
};

    
export default memo(Link);