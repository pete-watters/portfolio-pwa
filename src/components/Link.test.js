import React from 'react';
import { shallow } from 'enzyme';
import Link from './Link';

const linkMock = {
  href : 'https://pete.com',
  text : 'Test'
}

let wrapped = shallow(<Link href={linkMock.href} text={linkMock.text}/>);
describe('Title', () => {
  it('should render the Link Component correctly', () => {   
    expect(wrapped).toMatchSnapshot();
  });
  it('renders the Link text', () => { 
    expect(wrapped.find('a').text()).toEqual(linkMock.text);
    expect(wrapped.find('a').prop('href')).toEqual(linkMock.href);
  });
});