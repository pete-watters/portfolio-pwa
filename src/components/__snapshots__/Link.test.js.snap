// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title should render the Link Component correctly 1`] = `
<a
  href="https://pete.com"
  rel="noopener noreferrer"
  target="_blank"
>
  Test
</a>
`;
